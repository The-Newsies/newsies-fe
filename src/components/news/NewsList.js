import React from 'react';
import PropTypes from 'prop-types';
import NewsItem from './NewsItem';
import styles from './NewsList.css';
import Sentiment from 'sentiment';

export default function NewsItemsList({ news }) {
  const newsWithSentiment = news.map(article => {
    // sentiment analysis
    const sentiment = new Sentiment();
    const result = sentiment.analyze(article.title);
    const score = result.score;
    return { ...article, sentiment: score };
  });

  const sorted = newsWithSentiment.sort(function(a, b) {
    return b.sentiment - a.sentiment;
  });

  const newsList = sorted.map(({
    title,
    description,
    urlToImage,
    author,
    publishedAt,
    source,
    url
  }) => (
    <li key={url}>
      <NewsItem 
        title={title}
        description={description || 'no description'}
        urlToImage={urlToImage}
        author={author}
        publishedAt={publishedAt}
        source={source}
        url={url}
      />
    </li>
  ));

  return (
    <ul className={styles.container}>
      {newsList}
    </ul>
  );
}

NewsItemsList.propTypes = {
  news: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    description: PropTypes.string,
    urlToImage: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    publishedAt: PropTypes.string.isRequired,
    source: PropTypes.shape({ 
      name: PropTypes.string.isRequired,
      id: PropTypes.string
    }).isRequired,
    url: PropTypes.string.isRequired,
  })).isRequired
};
