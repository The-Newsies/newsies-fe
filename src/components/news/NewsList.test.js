import React from 'react';
import { shallow } from 'enzyme';
import NewsList from './NewsList';

describe('NewsList component', () => {
  it('renders NewsList', () => {
    const wrapper = shallow(<NewsList 
      news={[{
        title: 'Title',
        description: 'description',
        image: 'image url',
        author: 'author',
        date: 'date',
        sourceName: 'sourceName',
        sourceLink: 'sourceLink',
        tagColor: 'tagColor'
      }]}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
