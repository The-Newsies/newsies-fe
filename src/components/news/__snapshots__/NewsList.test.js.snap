// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsList component renders NewsList 1`] = `
<ul
  className="container"
>
  <li>
    <NewsItem
      author="author"
      description="description"
      tagColor="tagColor"
      title="Title"
    />
  </li>
</ul>
`;
